cmd/osquery-perf/agent.go

```diff
@@ -1546,18 +1558,13 @@ func (a *agent) config() error {
 			q.Query = m["query"].(string)
 
 			scheduledQueryName := packName + "_" + queryName
-			if existingEntry, ok := a.scheduledQueryData[scheduledQueryName]; ok {
-				// Keep lastRun if the query is already scheduled.
-				q.lastRun = existingEntry.lastRun
+			if lastRun, ok := existingLastRunData[scheduledQueryName]; ok {
+				q.lastRun = lastRun
 			}
-			scheduledQueryData[scheduledQueryName] = q
+			a.scheduledQueryData.Store(scheduledQueryName, q)
```

Comment:
logger goroutine could send an incomplete number of results (if it runs after `a.scheduledQueryData.Clear`), right?

Reply:
It could

Reply:
I've replaced the `sync.Map` with a pointer